commit 1ba4bca8342ac3d55bf29bdd8f622cd304e11816
Author: Kai Uwe Broulik <kde@privat.broulik.de>
Date:   Wed Jun 21 12:17:16 2017 +0200

    Introduce KDE_NO_GLOBAL_MENU env variable to disable global menu
    
    This allows to run specific Qt applications with global menu disabled in case of issues.
    
    Differential Revision: https://phabricator.kde.org/D6310

diff --git a/src/platformtheme/kdeplatformtheme.cpp b/src/platformtheme/kdeplatformtheme.cpp
index 994e1f2..1f026c0 100644
--- a/src/platformtheme/kdeplatformtheme.cpp
+++ b/src/platformtheme/kdeplatformtheme.cpp
@@ -64,6 +64,10 @@ static const QByteArray s_waylandAppMenuObjectPathPropertyName = QByteArrayLiter
 
 static bool checkDBusGlobalMenuAvailable()
 {
+    if (qEnvironmentVariableIsSet("KDE_NO_GLOBAL_MENU")) {
+        return false;
+    }
+
     QDBusConnection connection = QDBusConnection::sessionBus();
     QString registrarService = QStringLiteral("com.canonical.AppMenu.Registrar");
     return connection.interface()->isServiceRegistered(registrarService);
